spec/classes/openvmtools_init_spec.rb

```diff
@@ -2,293 +2,45 @@
-            minor: '0'
-          }
-        },
-        osfamily: 'RedHat',
-        virtual: 'vmware'
-      }
-    end
-
-    describe 'ensure => absent' do
-      let(:params) { { ensure: 'absent' } }
-
-      it { is_expected.to contain_package('open-vm-tools').with_ensure('absent') }
-      it { is_expected.to contain_service('vgauthd').with_ensure('stopped') }
-      it { is_expected.to contain_service('vmtoolsd').with_ensure('stopped') }
-    end
-
-    describe 'with_desktop => true' do
-      let(:params) { { with_desktop: true } }
-
-      it {
-        is_expected.to contain_package('open-vm-tools-desktop').with_ensure('present')
-      }
-    end
-
-    describe 'uninstall_vmware_tools => true' do
-      let(:params) { { uninstall_vmware_tools: true } }
-
-      it {
-        is_expected.to contain_package('VMwareTools').with_ensure('absent')
-      }
-    end
-  end
-
-  context 'on a supported FreeBSD 11 os, vmware platform, custom parameters' do
-    let :facts do
-      {
-        operatingsystem: 'FreeBSD',
-        operatingsystemmajrelease: '11',
-        operatingsystemrelease: '11.2-RELEASE',
-        os: {
-          family: 'FreeBSD',
-          name: 'FreeBSD',
-          release: {
-            full: '11.2-RELEASE',
-            major: '11',
-            minor: '2'
-          }
-        },
-        osfamily: 'FreeBSD',
-        virtual: 'vmware'
-      }
-    end
-
-    describe 'ensure => absent' do
-      let(:params) { { ensure: 'absent' } }
-
-      it { is_expected.to contain_package('open-vm-tools-nox11').with_ensure('absent') }
-      it { is_expected.to contain_service('vmware_guestd').with_ensure('stopped') }
-    end
-
-    describe 'with_desktop => true' do
-      let(:params) { { with_desktop: true } }
-
-      it { is_expected.to contain_package('open-vm-tools') }
-      it { is_expected.not_to contain_package('open-vm-tools-nox11') }
-
-      it {
-        is_expected.to contain_service('vmware_guestd').with(
-          ensure: 'running',
-          enable: true,
-          hasstatus: true,
-          require: '[Package[open-vm-tools]{:name=>"open-vm-tools"}]'
-        )
-      }
+describe 'openvmtools' do
+  on_supported_os.each do |os, os_facts|
+    context "os #{os}" do
+      let(:facts) do
+        os_facts.merge({ virtual: 'vmware' })
+      end
+      let(:service_name) do
+        case facts[:os]['family']
+        when 'Debian'
+          'open-vm-tools'
+        when 'FreeBSD'
+          'vmware_guestd'
+        else
+          'vmtoolsd'
+        end
+
+        it { is_expected.to compile.with_all_deps }
+
+        it { is_expected.to contain_package('open-vm-tools') }
+        it { is_expected.not_to contain_package('open-vm-tools-desktop') }
+
+        it {
+          is_expected.to contain_service('vgauthd').with(
+            ensure: 'running',
+            enable: true
+          )
+        }
+
+        it { is_expect.to contain_service('vgauthd').that_requires('Package[open-vm-tools]') }
```

Comment:
You can chain them, which typically helps to avoid compiling the catalog multiple times (speeding up tests):
```suggestion
        it {
          is_expected.to contain_service('vgauthd').with(
            ensure: 'running',
            enable: true
          ).that_requires('Package[open-vm-tools]')
        }
```

Reply:
I'm realize that vgauthd not exist on Debian based but on EL
can I pass an array to contain_service?